### ISSUE TITLE
Firefox latest version blocks request /co/authenticate (Custom domain activated)

### ISSUE THREAD

**Issue by akamal88** [auth0/auth0.js#1191]
### Describe the problem

> We have set up custom domain for our website auth.equeum.com and it's working perfect in all browsers except firefox, it's showing this error
![image](https://user-images.githubusercontent.com/37626525/124127732-c1639800-da84-11eb-8958-a2c470e474de.png)

### Reproduction

> Navigate into our website https://dev.equeum.com/login and try to login using any username/password
, watch the network tab, it will show status 504 Gateway Timeout.

### Environment

> Please provide the following:

- **Version of Auth0.js used:*9.16.2*
- **Which browsers have you tested in?*firefox 89.0.2*

**Comment by stevehobbsdev** [comment]
👋🏻 As you've mentioned the latest Firefox, does that imply that it used to work in previous versions of Firefox but has now broken in the latest version? Or has it never worked in Firefox?

**Comment by akamal88** [comment]
Yes, was working before in firefox, but wondering how we make it work again, now, i think the issue also related in how firefox treat with angular service worker but i am not 100% sure.
I hope someone had similar issue to help us fix the problem soon, Thanks a lot

**Comment by akamal88** [comment]
I was correct about angular service worker, look at the selected line 
![image](https://user-images.githubusercontent.com/37626525/124293517-6d79b180-db5f-11eb-9627-01dcfc119ba6.png)

i changed it from 
`var url = urlJoin(this.baseOptions.rootUrl, '/co/authenticate');`
to
`var url = urlJoin(this.baseOptions.rootUrl, '/co/authenticate?ngsw-bypass=true');`

then, everything works as expected, can you pls make a way to add query param to CrossOriginAuthentication?

**Comment by akamal88** [comment]
I created PR https://github.com/auth0/auth0.js/pull/1192

**Comment by stevehobbsdev** [comment]
So this is going to be a difficult one to solve quickly. I don't believe the proposed solution in #1192 is the right way forward as it's very specific to your scenario but affects everyone.

To start, I unfortunately can't reproduce the problem using the URL you provided as it has been modified to work.

More than that, the way the code is now we don't easily have a way that you can pass in custom query parameters to this end point. The request in question is a POST request that takes a JSON body, and we don't have any affordances that would allow you to pass in custom parameters here like some other endpoints, just because we're not expecting them. Normally we allow you to do that on some other login APIs as they can be used in custom rules, but I'm not sure the `/co/authenticate` endpoint uses Rules.

We also don't currently have a way for you to modify the request before it is sent.

Before I explore a complex & generic solution to help you here, is there a way this can be solved within your own application, perhaps in some configuration for the Angular service worker or some way you're able to prevent it from executing on certain calls? I'm not familiar with Angular service works so perhaps you can fill me in with what's possible.

**Comment by akamal88** [comment]
Hey Steve, unfortunately, I can't find a way to fix it using angular
service workers configuration.
we can set requests header instead of query parameters but the
authenticate/co OPTIONS restrict the incoming headers,  as a result it's
failing.
As a temp solution, I compiled the library with my modified changes and I
used it in my code.

Thanks,
Akram

في السبت، ١٧ يوليو ٢٠٢١ ١٢:٣٣ ص Steve Hobbs ***@***.***> كتب:
